Source/JavaScriptCore/jit/JITThunks.cpp

```diff
@@ -240,7 +252,7 @@ NativeExecutable* JITThunks::hostFunctionStub(VM& vm, TaggedNativeFunction funct
     
     Ref<JITCode> forConstruct = adoptRef(*new NativeJITCode(MacroAssemblerCodeRef<JSEntryPtrTag>::createSelfManagedCodeRef(ctiNativeConstruct(vm).retagged<JSEntryPtrTag>()), JITType::HostCallThunk, NoIntrinsic));
     
-    NativeExecutable* nativeExecutable = NativeExecutable::create(vm, forCall.releaseNonNull(), function, WTFMove(forConstruct), constructor, implementationVisibility, name);
+    NativeExecutable* nativeExecutable = NativeExecutable::create(vm, globalObject, forCall.releaseNonNull(), function, WTFMove(forConstruct), constructor, implementationVisibility, name);
```

Comment:
NativeExecutable is shared in multiple JSGlobalObject in one VM, so we cannot have a reference to JSGlobalObject, otherwise, it regress memory.

Reply:
ditto